lib/XMLHttpRequest.js

```diff
@@ -365,8 +410,21 @@ function XMLHttpRequest(opts) {
       throw new Error("INVALID_STATE_ERR: send has already been called");
     }
 
-    var ssl = false, local = false;
-    var url = Url.parse(settings.url);
+    var ssl = false, local = false, datauri = false;
+    var url;
+    try {
+      if (settings.origin != null) {
+        url = new URL(settings.url, settings.origin);
+      }
+      else {
+        url = new URL(settings.url);
+      }
+      settings.url = url.href;
+    }
+    catch (e) {
+      self.handleError(new Error("ERR_INVALID_URL"));
```

Comment:
Why not handle the original error `e` instead?

Reply:
browsers usually return "ERR_INVALID_URL" if the given url is invalid, specifying the error in parsing URL is not that necessary imo.

Reply:
This is actually the exact error that will be thrown, hence my comment. However, your error has an `"ERR_INVALID_URL"` `message`, while the actual error has it as a `code`.

Reply:
For URL constructor, it throws a message like "Invalid Base URL" or "Invalid URL", I could switch to throwing that message instead.

Yeah, this one could be helpful in case the user defines an invalid origin (so user may have more idea what's going wrong with their URL).

Reply:
```js
> try { new URL("").href } catch (e) {
    console.log(e.message);
}

< Failed to construct 'URL': Invalid base URL
```
However, this is the original message after it throws, should i just either keep the last part (Invalid URL) or just keep the whole message?